database/migrations/2023_12_19_211314_user_unique_email.php

```diff
@@ -4,12 +4,21 @@
 use Illuminate\Database\Schema\Blueprint;
 use Illuminate\Support\Facades\Schema;
 
+require_once 'include/common.php';
+
 return new class extends Migration {
     /**
      * Run the migrations.
      */
     public function up(): void
     {
+        // Delete any duplicate email addresses, keeping the oldest userid for each one.
+        $user_table =  qid('user');
+        $num_deleted = DB::delete(
+            "DELETE FROM $user_table WHERE id NOT IN (SELECT * FROM (SELECT MIN(id) FROM $user_table GROUP BY email) AS A)");
```

Comment:
This could probably be done more efficiently as well, but that shouldn't matter given the size of the users table for even the largest systems.